inductiva/templating/renderers.py

```diff
@@ -0,0 +1,120 @@
+"""Base and specific renderers packages to manage template files."""
+from abc import abstractmethod
+from typing import Iterable, Union, IO
+import pathlib
+import shutil
+import os
+
+import jinja2
+
+from inductiva import types, utils
+
+
+class BaseRenderer:
+    """Abstract base class for template renderers."""
+
+    TEMPLATE_EXTENSION = None
+
+    @abstractmethod
+    def render_dir(self, target_dir: types.Path, source_dir: types.Path,
+                   **render_args):
+        """Render a template to a file."""
+        pass
+
+    @abstractmethod
+    def render_file(self, target_file: types.Path,
+                    source_file: Union[types.Path, IO], **render_args):
+        """Render a file."""
+        pass
+
+    @classmethod
+    def is_template(cls, file: types.Path) -> bool:
+        """Check if the given file is of template type."""
+        if isinstance(file, pathlib.Path):
+            return file.suffix == cls.TEMPLATE_EXTENSION
+        return file.endswith(cls.TEMPLATE_EXTENSION)
+
+    @classmethod
+    def strip_extension(cls, file: types.Path) -> str:
+        """Strip the template extension if the given file is a template."""
+        if not cls.is_template(file):
+            return file
+        if isinstance(file, pathlib.Path):
+            return file.with_suffix("")
+        return os.path.splitext(file)[0]
+
+    @classmethod
+    def get_template_files(cls, source_dir: types.Path) -> Iterable[types.Path]:
+        """Return a list of template files in the given source directory."""
+
+        return utils.files.map_dir_files(source_dir, cls.TEMPLATE_EXTENSION)
+
```

Comment:
I don't like this method here, nor do I like having this `map_dir_files` function in utils. It should be part of the file manager.

Reply:
Can you elaborate? I felt that this was a general function, so there wasn't a reason to attach it to the file_manager. Moreover, this is used in a different manner inside the TemplateEngine.